code/modules/mob/living/simple_animal/hostile/headcrab.dm

```diff
@@ -191,3 +198,363 @@
+		return
+
+	//out of combat
+	if(homesick >= gonarch_rampage_trigger && mode != GONARCH_MODE_RAMPAGE)
+		mode_switch(GONARCH_MODE_RAMPAGE, gonarch_soundlist_rampage, change_icon = "screech_pose")
+		return
+	switch(mode)
+		if(GONARCH_MODE_REST)		// idle
+			gonarch_rest()
+		if(GONARCH_MODE_RETURNING)		// returning home
+			if(world.time > gonarch_rangeattack_swoop_lastattack + gonarch_rangeattack_swoop_interval)
+				shoot_swoop()
+			gonarch_return()
+		if(GONARCH_MODE_NEWHOME)		// Seeking new home
+			gonarch_newhome()
+		if(GONARCH_MODE_RAMPAGE)		// Cannot find nest, frustration at max - rampage
+			gonarch_rampage()
+		if(GONARCH_MODE_DEFENDING) //If it goes down here, there is no target anymore. And it should return.
+			mode_switch(GONARCH_MODE_RETURNING, gonarch_soundlist_return, change_icon = "gonarch_peaceful")
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/shoot_artillery(target)
+	gonarch_rangeattack_lastattack = world.time
+	for(var/turf/T in range(gonarch_rangeattack_range, target))
+		if(isspaceturf(T))
+			continue
+		if(prob(gonarch_rangeattack_turfchance))
+			addtimer(CALLBACK(src, .proc/shoot_artillery_impact, T), rand(5, 35))
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/shoot_artillery_impact(target_turf)
+	new /obj/effect/temp_visual/gonarch_artillery(target_turf)
+
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/shoot_swoop()
+	gonarch_rangeattack_swoop_lastattack = world.time
+	for(var/mob/living/carbon/C in range(gonarch_rangeattack_swoop_range, src))
+		addtimer(CALLBACK(src, .proc/shoot_projectile, C.loc), rand(5, 35))
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/shoot_projectile(turf/marker, set_angle)
+	if(!isnum(set_angle) && (!marker || marker == loc))
+		return
+	var/turf/startloc = get_turf(src)
+	var/obj/item/projectile/P = new /obj/item/projectile/gonarch(startloc)
+	P.preparePixelProjectile(marker, marker, startloc)
+	P.firer = src
+	if(target)
+		P.original = target
+	P.fire(set_angle)
+
+/obj/item/projectile/gonarch
+	name ="acid bolt"
+	icon_state= "chronobolt"
+	damage = 10
+	armour_penetration = 0
+	speed = 2
+	eyeblur = 2
+	damage_type = BURN
+	pass_flags = PASSTABLE
+	/// Used to define in what range the splash damage will be applied to.
+	var/splash_range = 1
+
+/obj/item/projectile/gonarch/on_hit(atom/target, blocked = 0)
+	. = ..()
+	if(iscarbon(target))
+		for(var/mob/living/C in range(splash_range, target))
+			if(istype(C, /mob/living/simple_animal/hostile/headcrab))
+				continue
+			C.adjustFireLoss(10)
+			C.adjustToxLoss(10)
+			if(C.reagents)
+				C.reagents.add_reagent("sacid", 10)
+				to_chat(C, "<span class='userdanger'>You have been splashed with acid!</span>")
+
+/obj/effect/temp_visual/gonarch_artillery
+	icon = 'icons/mob/actions/actions.dmi'
+	icon_state = "sniper_zoom"
+	layer = BELOW_MOB_LAYER
+	light_range = 2
+	light_color = "#000000"
+	duration = 9
+
+/obj/effect/temp_visual/gonarch_artillery/ex_act()
```

Comment:
Not required. `/obj/effect/temp_visual/ex_act()` already returns and does nothing